### PR TITLE
Add article about upcoming Kubernetes API removals

### PR DESCRIPTION
**Pull request by sftim**
Add a blog article [[preview](https://deploy-preview-28694--kubernetes-io-main-staging.netlify.app/blog/2021/07/14/upcoming-changes-in-kubernetes-1-22/)] to describe deprecated Kubernetes APIs that are being removed in v1.22

- relevant to https://github.com/kubernetes/sig-release/discussions/1606 (specifically for v1.22)
- relevant to https://github.com/kubernetes/website/pull/28182

/language en
/priority important-soon

### CODE REVIEW

**Comment by sftim** [comment]
@skrishna-unix provided you've [signed the CLA](https://kubernetes.io/docs/contribute/new-content/overview/#sign-the-cla), you can use the GitHub [suggestion](https://docs.github.com/en/github/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request) feature to propose changes directly.

**Comment by sftim** [comment]
@chrisnegus is there anything you suggest adding / changing here?

**Comment by PI-Victor** [comment]
i'll have a look at this too.

/assign

**Comment by sftim** [comment]
@skrishna-unix how's this looking?

**Comment by sftim** [comment]
@skrishna-unix it looks like you haven't [signed the CLA](https://kubernetes.io/docs/contribute/new-content/overview/#sign-the-cla). If you'd like to contribute text to this article, please do sign that either using a corporate CLA or the individual option.

**Comment by sftim** [comment]
/hold
https://github.com/kubernetes/website/pull/28694#issuecomment-876805629

**Comment by skrishna-unix** [comment]
> @skrishna-unix it looks like you haven't [signed the CLA](https://kubernetes.io/docs/contribute/new-content/overview/#sign-the-cla). If you'd like to contribute text to this article, please do sign that either using a corporate CLA or the individual option.

Sorry, Thanks for the reminder Tim, Just singed the CLA.

**Comment by skrishna-unix** [comment]
> @skrishna-unix how's this looking?

lgtm

**Comment by sftim** [comment]
but!
/hold
again

let's get the publication date right and sync wording with the v1.22 release team

**Review by mbevc1 — APPROVED** [review]

File: content/en/blog/_posts/2021-07-01-upcoming-changes-in-kubernetes-1-22/index.md
Comment (cirulls) [inline content/en/blog/_posts/2021-07-01-upcoming-changes-in-kubernetes-1-22/index.md:125]:
```suggestion
version. For example, if you have a manifest in source control that uses the beta
```

File: content/en/blog/_posts/2021-07-01-upcoming-changes-in-kubernetes-1-22/index.md
Comment (cirulls) [inline content/en/blog/_posts/2021-07-01-upcoming-changes-in-kubernetes-1-22/index.md:129]:
```suggestion
API fields remain the same. However, migrating other resources (e.g. deployments)
```

File: content/en/blog/_posts/2021-07-01-upcoming-changes-in-kubernetes-1-22/index.md
Comment (cirulls) [inline content/en/blog/_posts/2021-07-01-upcoming-changes-in-kubernetes-1-22/index.md:133]:
```suggestion
will require some updates based on changed fields. You can use the
`kubectl convert` command to automatically convert an existing object.
For example, to convert
```

Comment (cirulls) [reply content/en/blog/_posts/2021-07-01-upcoming-changes-in-kubernetes-1-22/index.md:133]: This is currently displayed like this:

image

File: content/en/blog/_posts/2021-07-01-upcoming-changes-in-kubernetes-1-22/index.md
Comment (cirulls) [inline content/en/blog/_posts/2021-07-01-upcoming-changes-in-kubernetes-1-22/index.md:142]:
```suggestion
defaults, etc.
```

File: content/en/blog/_posts/2021-07-01-upcoming-changes-in-kubernetes-1-22/index.md
Comment (cirulls) [inline content/en/blog/_posts/2021-07-01-upcoming-changes-in-kubernetes-1-22/index.md:147]:
```suggestion
removals before you upgrade to Kubernetes v1.22.
```

File: content/en/blog/_posts/2021-07-01-upcoming-changes-in-kubernetes-1-22/index.md
Comment (cirulls) [inline content/en/blog/_posts/2021-07-01-upcoming-changes-in-kubernetes-1-22/index.md:155]:
```suggestion
tools, custom controllers, etc.) still work how you expect, and you can revert if
```

Comment (cirulls) [reply content/en/blog/_posts/2021-07-01-upcoming-changes-in-kubernetes-1-22/index.md:155]: Nitpick: the comma before `etc.` is usually required as it is a series (see [here](https://stptrans.com/style-guide/english-language-guidelines/) under section Comma and [here](https://en.wikipedia.org/wiki/Serial_comma)). However, the website may use a different style so feel free to ignore it if it's not consistent with the current style guide.

File: content/en/blog/_posts/2021-07-01-upcoming-changes-in-kubernetes-1-22/index.md
Comment (cirulls) [inline content/en/blog/_posts/2021-07-01-upcoming-changes-in-kubernetes-1-22/index.md:168]:
```suggestion
option set through and after the upgrade to v1.22.
```

**Review by cirulls — COMMENTED** [review]
This is a very good and detailed blog article and will be very helpful for migration to v1.22. 

I mainly reviewed the article from an editorial point of view as I'm not a subject expert in the topic. I suggested few editorial changes, mainly nitpicks. I checked that all the links resolve and are relevant.

**Comment by strongjz** [comment]
lgtm , very detailed on all the API changes, thanks for putting this together.

**Comment by sftim** [comment]
Date is pencilled in
/hold cancel

**Review by onlydole — APPROVED** [review]
@skrishna-unix and @sftim, this looks fantastic! Thank you for working on this, and @cirulls for your feedback.

File: content/en/blog/_posts/2021-07-14-upcoming-changes-in-kubernetes-1-22/index.md
Comment (liggitt) [inline content/en/blog/_posts/2021-07-14-upcoming-changes-in-kubernetes-1-22/index.md:45]: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-22 has a lot of details about the exact changes between the beta and v1 versions... it might be good to link to that content or hoist it here

File: content/en/blog/_posts/2021-07-14-upcoming-changes-in-kubernetes-1-22/index.md
Comment (liggitt) [inline content/en/blog/_posts/2021-07-14-upcoming-changes-in-kubernetes-1-22/index.md:163]: this is not accurate... `kube-apiserver` still speaks to webhooks using the same API version as before by default... there's no change in 1.22

File: content/en/blog/_posts/2021-07-14-upcoming-changes-in-kubernetes-1-22/index.md
Comment (liggitt) [inline content/en/blog/_posts/2021-07-14-upcoming-changes-in-kubernetes-1-22/index.md:102]: The `kube-apiserver` default value for `--authentication-token-webhook-version` remains `v1beta1` in v1.22.

What changed in `kube-apiserver` in 1.22 is that it no longer **serves** the v1beta1 TokenReview API.

File: content/en/blog/_posts/2021-07-14-upcoming-changes-in-kubernetes-1-22/index.md
Comment (liggitt) [inline content/en/blog/_posts/2021-07-14-upcoming-changes-in-kubernetes-1-22/index.md:153]: should be:
```
...,networking.k8s.io/v1beta1=false
```

(disable the entire networking.k8s.io/v1beta1 group/version, rather than disabling individual resources)

File: content/en/blog/_posts/2021-07-14-upcoming-changes-in-kubernetes-1-22/index.md
Comment (sftim) [reply content/en/blog/_posts/2021-07-14-upcoming-changes-in-kubernetes-1-22/index.md:163]: Oh! That'll need updating.

File: content/en/blog/_posts/2021-07-14-upcoming-changes-in-kubernetes-1-22/index.md
Comment (sftim) [reply content/en/blog/_posts/2021-07-14-upcoming-changes-in-kubernetes-1-22/index.md:153]: Won't that also disable v1beta1 of EndpointSlice?

Comment (liggitt) [reply content/en/blog/_posts/2021-07-14-upcoming-changes-in-kubernetes-1-22/index.md:153]: that's a good point, it will. regardless, individual resource-level disablement isn't supported for any groups other than extensions/v1beta1

File: content/en/blog/_posts/2021-07-14-upcoming-changes-in-kubernetes-1-22/index.md
Comment (liggitt) [reply content/en/blog/_posts/2021-07-14-upcoming-changes-in-kubernetes-1-22/index.md:45]: I'd probably link to https://kubernetes.io/docs/reference/using-api/deprecation-guide/ rather than duplicate a ton of content, so we have a non-blog authoritative destination page for all those details

Comment (sftim) [reply content/en/blog/_posts/2021-07-14-upcoming-changes-in-kubernetes-1-22/index.md:45]: Maybe we could make a task page about updating to Ingress v1, as that's more detailed in the blog than in https://kubernetes.io/docs/reference/using-api/deprecation-guide/

However, I'm not sure where we'd find the people / time to add that task. I will hyperlink to that deprecation guide.

File: content/en/blog/_posts/2021-07-14-upcoming-changes-in-kubernetes-1-22/index.md
Comment (sftim) [reply content/en/blog/_posts/2021-07-14-upcoming-changes-in-kubernetes-1-22/index.md:153]: Ah. I didn't test this. So actually there's no way to simulate exactly this removal. I'll call that out.

**Pull request merged**
